include/swift/AST/Stmt.h

```diff
@@ -1072,7 +983,9 @@ class CaseStmt final
   }
   SourceLoc getEndLoc() const { return getBody()->getEndLoc(); }
   SourceRange getLabelItemsRange() const {
-    return ColonLoc.isValid() ? SourceRange(getLoc(), ColonLoc) : getSourceRange();
+    return ItemTerminatorLoc.isValid()
+               ? SourceRange(getLoc(), ItemTerminatorLoc)
+               : getSourceRange();
```

Comment:
This includes `{` loc for `catch` statement. Considering the current usages, I don't think that affects anything, but I feel like it shouldn't. But this doesn't block this from merging though.

Reply:
Good point, I hadn't considered that but it was inconsistent with how ranges work for other statements. I've updated this so the range now extends from `case` to the `:` in switch cases, and from `catch` to the end of the last item in catch clauses, so the `{` is excluded. For `catch {...}`, the range now only covers `catch`.